src/api/File.js

```diff
@@ -78,6 +82,56 @@ class File extends Item {
         this.successHandler(`${downloadBaseUrl}?${downloadUrlQuery}`);
     }
 
+    /**
+     * Polls a representation's infoUrl, attempting to generate a representation
+     *
+     * @param {FileRepresentation} representation - representation that should have it's info.url polled
+     * @return {Promise<FileRepresentation | Error>} - representation updated with most current status or Error object
+     */
+    async generateRepresentation(representation: FileRepresentation): Promise<FileRepresentation | Error> {
+        const infoUrl = getProp(representation, 'info.url');
+
+        const numOfTries = 4;
+        const initialTimeout = 2000;
+        const backoffFactor = 2;
+
+        if (!infoUrl) {
+            return representation;
+        }
+
+        const isStatusSuccess = response => {
+            const status = getProp(response, 'data.status.state');
+            return (
+                !status ||
+                status === REPRESENTATIONS_RESPONSE_ERROR ||
+                status === REPRESENTATIONS_RESPONSE_SUCCESS ||
+                status === REPRESENTATIONS_RESPONSE_VIEWABLE
+            );
+        };
+
+        return this.xhr
+            .get({ url: infoUrl })
+            .then(() =>
+                retryNumOfTimes(
+                    (successCallback, errorCallback) =>
+                        this.xhr
+                            .get({ url: infoUrl })
+                            .then(response =>
+                                isStatusSuccess(response)
+                                    ? successCallback(response.data)
+                                    : errorCallback(response.data),
+                            )
+                            .catch(e => {
+                                errorCallback(e);
+                            }),
+                    numOfTries,
+                    initialTimeout,
+                    backoffFactor,
+                ),
+            )
+            .catch(e => e);
```

Comment:
I think I was mistaken here, we should probably just let the promise rejection return as normal instead of returning it as a promise